src/bentoml/serve.py

```diff
@@ -500,12 +527,41 @@ def serve_http_production(
             f"{scheme}://{log_host}:{port}/metrics",
         )
 
-    arbiter = create_standalone_arbiter(
-        watchers=watchers,
-        sockets=list(circus_socket_map.values()),
-    )
+    arbiter_kwargs: dict[str, t.Any] = {
+        "watchers": watchers,
+        "sockets": list(circus_socket_map.values()),
+    }
+    if development_mode:
+        plugins = []
+        if reload:
+            if sys.platform == "win32":
+                logger.warning(
+                    "Due to circus limitations, output from the reloader plugin will not be shown on Windows."
+                )
+            logger.debug(
+                "--reload is passed. BentoML will watch file changes based on 'bentofile.yaml' and '.bentoignore' respectively."
+            )
+
+            # NOTE: {} is faster than dict()
+            plugins = [
+                # reloader plugin
+                {
+                    "use": "bentoml._internal.utils.circus.watchfilesplugin.ServiceReloaderPlugin",
+                    "working_dir": working_dir,
+                    "bentoml_home": bentoml_home,
+                },
+            ]
```

Comment:
Let's allow people to specify reload outside of development mode, I think that's fine.

Reply:
I have test reload for production server and seems working ok. So now I apply `--reload` for both production and development.